src/plugins/GUITestBase/src/tests/common_scenarios/msa_editor/edit/GTTestsMSAEditorEdit.cpp

```diff
@@ -864,5 +864,53 @@ GUI_TEST_CLASS_DEFINITION(test_0015) {
     GTUtilsMSAEditorSequenceArea::callContextMenu(os);
 }
 
+GUI_TEST_CLASS_DEFINITION(test_0016_1) {
+    // 1. Open document "data/samples/CLUSTALW/COI.aln".
+    GTFileDialog::openFile(os, dataDir + "samples/CLUSTALW/COI.aln");
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+
+    static constexpr QPoint TOP_LEFT = QPoint(0, 0);
+    static constexpr QPoint BOTTOM_RIGHT = QPoint(10, 10);
+
+    //2. Select area from (0, 0) to (10, 10)
+    GTUtilsMSAEditorSequenceArea::selectArea(os, TOP_LEFT, BOTTOM_RIGHT);
+
+    // 3. Click "Replace with gaps" with popup menu
+    GTUtilsDialog::waitForDialog(os, new PopupChooser(os, { "MSAE_MENU_EDIT", "replace_with_gaps" }));
+    GTMenu::showContextMenu(os, GTWidget::findWidget(os, "msa_editor_sequence_area"));
+
+    // Expected: selection hasn't been changed
+    auto selectedRect = GTUtilsMSAEditorSequenceArea::getSelectedRect(os);
+    auto tl = selectedRect.topLeft();
+    CHECK_SET_ERR(tl == TOP_LEFT, QString("Expected top-left selection: 0, 0; current: %1, %2").arg(tl.x()).arg(tl.y()));
+
+    auto br = selectedRect.bottomRight();
+    CHECK_SET_ERR(br == BOTTOM_RIGHT, QString("Expected bottom-right selection: 0, 0; current: %1, %2").arg(br.x()).arg(br.y()));
+}
+
+GUI_TEST_CLASS_DEFINITION(test_0016_2) {
+    // 1. Open document "data/samples/CLUSTALW/COI.aln".
+    GTFileDialog::openFile(os, dataDir + "samples/CLUSTALW/COI.aln");
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+
+    static constexpr QPoint TOP_LEFT = QPoint(0, 0);
+    static constexpr QPoint BOTTOM_RIGHT = QPoint(10, 10);
+
+    // 2. Select area from (0, 0) to (10, 10)
+    GTUtilsMSAEditorSequenceArea::selectArea(os, TOP_LEFT, BOTTOM_RIGHT);
+
+    // 3. Click "Replace with gaps" with shortcut
+    GTKeyboardDriver::keyClick(Qt::Key_Space, Qt::ControlModifier);
+
+    // Expected: selection hasn't been changed
+    auto selectedRect = GTUtilsMSAEditorSequenceArea::getSelectedRect(os);
+    auto tl = selectedRect.topLeft();
+    CHECK_SET_ERR(tl == TOP_LEFT, QString("Expected top-left selection: 0, 0; current: %1, %2").arg(tl.x()).arg(tl.y()));
+
+    auto br = selectedRect.bottomRight();
+    CHECK_SET_ERR(br == BOTTOM_RIGHT, QString("Expected bottom-right selection: 0, 0; current: %1, %2").arg(br.x()).arg(br.y()));
```

Comment:
Make the same (content) check here. Otherwise, your test will pass even if you do not press any key at all.

Reply:
OK

Reply:
You have 2 tests above: Space and Shift+Space.

The problem with these tests is that if I replace Space -> Shift+Space or Shift+Space to Space they both will work: because they do not test the difference between each other's behavior.